src/bin/pgcopydb/extensions.c

```diff
@@ -407,6 +407,39 @@ copydb_parse_extensions_requirements(CopyDataSpec *copySpecs, char *filename)
 }
 
 
+/* source_has_timescaledb_extension will check if timescaledb extension is present in 
+ * source database, if it exists then prepare the atrget database for timescale data
+ */
+bool 
+source_has_timescaledb_extension(CopyDataSpec *copySpecs, bool *hasTimescaledb)
```

Comment:
I think @dimitri meant to query the sqlite catalog not the source again.

Reply:
Yes, I misunderstood it earlier. I have now implemented the lookup function to query the SQLite catalog instead of the source. Thank you for pointing that out!